server/game/cards/Time/Generosity.js

```diff
@@ -0,0 +1,84 @@
+const { CardType, BattlefieldTypes } = require('../../../constants.js');
+const Card = require('../../Card.js');
+
+class Generosity extends Card {
+    setupCardAbilities(ability) {
+        this.play({
+            title: 'Generosity',
+            effect: 'choose from 3 actions',
+            gameAction: ability.actions.chooseAction((context) => ({
+                target: context.player,
+                choices: this.getChoices(ability, context.player, context)
+            })),
+            then: {
+                alwaysTriggers: true,
+                gameAction: ability.actions.chooseAction((context) => ({
+                    target: context.player.opponent,
```

Comment:
not totally sure if both of these player and target are required. i know at least one is. felt better to just be safe

Reply:
i can test and remove the unnecessary ones though

Reply:
turns out target is always required. player is only required for the opponent's choice. so jsut that 1 player for the third action can be removed